tests/std/tests/GH_003570_allocate_at_least/test.cpp

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <cassert>
+#include <deque>
+#include <memory>
+#include <sstream>
+#include <string>
+#include <syncstream>
+#include <utility>
+#include <vector>
+
+using namespace std;
+
+struct {
+    void assert_set() {
+        assert(consume());
+    }
+
+    bool consume() {
+        return exchange(is_set, false);
+    }
+
+    void set() {
+        is_set = true;
+    }
+
+private:
+    bool is_set = false;
+} allocate_at_least_signal;
+
+template <class T>
+struct signalling_allocator {
+    using value_type = T;
+
+    signalling_allocator() = default;
+
+    template <typename U>
+    signalling_allocator(const signalling_allocator<U>&) {}
+
+    T* allocate(size_t count) {
+        T* ptr = (T*) malloc(count * sizeof(T));

```

Comment:
Style: We conventionally avoid C casts in both product and test code, except `(void)`.